### PR TITLE
Fixes #127: Added support for file attachments

### LINKED ISSUE #124: If the user have got a 403 on the bucket, it cannot open the collection.
403-on-bucket

error-undefined

### LINKED ISSUE #127: Add support for file attachments

### PR DESCRIPTION
**Pull request by n1k0**
Refs #127. Note that this also fixes #124 and #126.

#### Todo

- [x] Add fields for enabling and configuring file attachments in CollectionForm
- [x] Add a checkbox to specify if attachments are required or optional
- [x] Update RecordForm to render a file widget when attachments are enabled for the collection
- [x] Add checks to expose the feature only if the server capability is available
- [x] On record edition form, render information about current attachment, if any
- [x] Allow deletion of an existing attachment
- [x] Bulk form support
- [x] Add tests

### CODE REVIEW

**Comment by n1k0** [comment]
This is now ready for review, and the patch has been deployed to gh-pages.

r=? @Natim @leplatrem @glasserc

**Comment by Natim** [comment]
I tried it, it looks great thanks a lot for this.

File: scripts/actions/notifications.js [hunk 1]
```diff
@@ -28,9 +28,7 @@ export function notifySuccess(message, options) {
 }
 
 export function notifyError(error, options) {
-  if (process.env.NODE_ENV !== "production") {
-    console.error(error);
-  }
+  console.error(error);
```
Comment (Natim) [inline scripts/actions/notifications.js]: Is it what you actually want?

File: scripts/client.js [hunk 2]
```diff
@@ -53,3 +38,27 @@ export function setClient(_client) {
 export function resetClient() {
   client = null;
 }
+
+// XXX this should eventually move to kinto-client
```
Comment (Natim) [inline scripts/client.js]: Indeed, but it is good enough here for now.

File: scripts/components/CollectionEdit.js [hunk 3]
```diff
@@ -29,9 +29,9 @@ function DeleteForm({cid, onSubmit}) {
 
 export default class CollectionEdit extends Component {
   onSubmit = (formData) => {
-    const {params} = this.props;
+    const {params, updateCollection} = this.props;
     const {bid, cid} = params;
-    this.props.updateCollection(bid, cid, formData);
+    updateCollection(bid, cid, formData);
```
Comment (Natim) [inline scripts/components/CollectionEdit.js]: yeah cleaning :)

File: scripts/actions/notifications.js (hunk 1 above)
Comment (n1k0) [inline scripts/actions/notifications.js]: Yes, I realized that reporting bugs is a pain if users can't simply inspect the console and tell us what went wrong precisely.

**Comment by Natim** [comment]
Seems good to me.

**Comment by Natim** [comment]
r=

**Pull request merged**